### PR TITLE
Use if-let or when-let when applicable

### PR DESCRIPTION
**Pull request by selesdepselesnul**

### CODE REVIEW

File: events.lisp [hunk 1]
```diff
@@ -521,37 +515,34 @@ converted to an atom is removed."
...
-               (case (safe-atom-name p)
-                 (:_NET_WM_STATE_DEMANDS_ATTENTION
-                  (case action
-                    (1
-                     (add-wm-state window :_NET_WM_STATE_DEMANDS_ATTENTION))
-                    (2
-                     (unless (find-wm-state window :_NET_WM_STATE_DEMANDS_ATTENTION)
-                       (add-wm-state window :_NET_WM_STATE_DEMANDS_ATTENTION))))
-                  (maybe-set-urgency our-window))
+     (when-let ((our-window (find-window window)))
```
Comment (PuercoPop) [inline events.lisp]: Can we hoist the bindings from the inner `let` to the `when-let`

File: events.lisp [hunk 2]
```diff
@@ -521,37 +515,34 @@ converted to an atom is removed."
...
+                (update-fullscreen our-window action))))))))
   (:_NET_MOVERESIZE_WINDOW
-   (let ((our-window (find-window window)))
-     (when our-window
-       (let ((x (elt data 1))
-             (y (elt data 2)))
-         (dformat 3 "!!! Data: ~S~%" data)
-         (group-move-request (window-group our-window) our-window x y :root)))))
+   (when-let ((our-window (find-window window)))
+     (let ((x (elt data 1))
```
Comment (PuercoPop) [inline events.lisp]: ditto

File: events.lisp [hunk 3]
```diff
@@ -152,29 +150,26 @@
   (dformat 2 "UNMAP: ~s ~s ~a~%" send-event-p (not (xlib:window-equal event-window window)) (find-window window))
   (unless (and (not send-event-p)
                (not (xlib:window-equal event-window window)))
-    (let ((window (find-window window)))
+    (when-let ((window (find-window window)))
       ;; if we can't find the window then there's nothing we need to
```
Comment (PuercoPop) [inline events.lisp]: The comment would now belong on top of the when-let, but I think we can remove it altogether. It is redundant with the code.

**Comment by selesdepselesnul** [comment]
done :smile:

**Comment by PuercoPop** [comment]
@selesdepselesnul Awesome, will merge it when I get home. Thanks for all the recent cleanups you've been doing

File: events.lisp [hunk 4]
```diff
@@ -152,29 +150,24 @@
   (dformat 2 "UNMAP: ~s ~s ~a~%" send-event-p (not (xlib:window-equal event-window window)) (find-window window))
   (unless (and (not send-event-p)
                (not (xlib:window-equal event-window window)))
-    (let ((window (find-window window)))
-      ;; if we can't find the window then there's nothing we need to
-      ;; do.
-      (when window
```
Comment (dbjergaard) [inline events.lisp]: Please be careful of obliterating comments when you are cleaning up constructions.

**Comment by dbjergaard** [comment]
@selesdepselesnul Your recent PRs are really showing some TLC that stumpwm badly needs! Could you kindly restore the comment that got deleted in one of the refactoring edits? I know the comment may seem to state the obvious, but sometimes what is obvious to some isn't to others.

If you are looking for other places to contribute, the code in the contrib modules is in very poor shape.  Some of them are bitrotting away and others just need documentation or a link in the main readme.

**Comment by selesdepselesnul** [comment]
done :smile:

**Pull request merged**